src/FixedDataTable.js

```diff
@@ -920,7 +920,7 @@ var FixedDataTable = React.createClass({
     }
 
     var lastScrollTop = oldState ? oldState.scrollTop : undefined;
-    if (props.scrollTop !== lastScrollTop) {
+    if (typeof props.scrollTop !== 'undefined' && props.scrollTop !== null && props.scrollTop !== lastScrollTop) {
```

Comment:
Cool, I think the case you're referring to is specific to references on the execution context (props for this case), rather than accessed through an object with dot notation scrollTop.

If there's a risk of props not existing, we'd need to also add a `typeof props !== undefined &&` check to avoid an error from the dot access.  If not, I think we're safe with the short hand Kam suggests in both strict and non-strict modes.